File: Baekjoon/DP/1010.py
# https://www.acmicpc.net/problem/1010

t = int(input())
dp = [[0] * 30 for _ in range(30)]

for i in range(30):
    for j in range(30):
        if i == 1:
            dp[i][j] = j
        else:
            if i == j:
                dp[i][j] = 1
            elif i < j:
                dp[i][j] = dp[i - 1][j - 1] + dp[i][j - 1]

for _ in range(t):
    n, m = list(map(int, input().split()))
    print (dp[n][m])

File: Baekjoon/DP/1010.cpp
// https://www.acmicpc.net/problem/1010

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n, m;
        cin >> n >> m;

        vector<vector<int>> dp(n + 1, vector<int>(m + 1, 0));
        for (int i = 1; i <= m; i++) {
            dp[1][i] = i;
        }

        for (int i = 2; i <= n; i++) {
            for (int j = i; j <= m; j++) {
                for (int k = j; k >= i; k--) {
                    dp[i][j] += dp[i - 1][k - 1];
                }
            }
        }

        cout << dp[n][m] << endl;
    }

    return 0;
}